internal/server/proto/server.proto

```diff
@@ -287,6 +287,53 @@ message Application {
   string file_change_signal = 3;
 }
 
+// Variable stores a variable's value, and metadata to allow for precedence 
+// sorting and source-specific error messaging 
+message Variable {
+  string name = 1;
+
+  oneof value {
+    string str = 2;
+    string hcl = 3;
+  }
+
+  // Source tracks where a value came from for more helpful error messaging.
+  // This is optional. If unset, we will give a generic error message.
+  oneof source {
+    SourceCLIArg cliarg = 4;
+    SourceCLIFile file = 5;
+    SourceEnvVar envvar= 6;
+    SourceRepoFile repofile = 7;
+    SourceServer server = 8;
+  }
+
+  // SourceCLIArg is set if a variable value is set via a -var flag
+  message SourceCLIArg {}
+
+  // SourceCLIFile is set if a variable value is set via a -var-file flag
+  message SourceCLIFile {
+    string fileName = 1;
+    bytes hclPos = 2;
```

Comment:
What do you expect these `bytes` to be?

In my mind, we'd probably want to recreate the `hcl.Pos` and/or `hcl.Range` types in proto: https://pkg.go.dev/github.com/hashicorp/hcl/v2#Pos And use that structurally. It should be relatively easy to get these out of HCL.

Reply:
I was expecting the `bytes` to be the Pos/Range data, yes, and abstractly was assuming I'd convert those to the HCL type once I got the bytes out of the proto message. By "recreate in proto" do you mean create a new message type? Cause that sounds good too.